flaskr/__init__.py

```diff
@@ -84,28 +84,57 @@ def index():
 				return render_template('index.html', stock_dict=session['stock_dict'])
 			return render_template('index.html', stock_dict=None)
 
-	# Results page (generates session id or code)
+	# Compare page
+	@app.route('/compare', methods=['GET', 'POST'])
+	def compare():
+		if request.method == 'POST':
+			uid1 = request.form['person1']
+			uid2 = request.form['person2']
+			error = None
+
+			if uid1 == uid2:
+				error = "IDs cannot be the same!"
+
+			elif db.get_profile(uid1)==None or db.get_profile(uid2)==None:
+				error = "Invalid ID"
+
+			#TODO: result generation and presentation
+			if error is None:
+				return render_template('results.html')
+			else:
+				flash(error)
+				return render_template('compare.html')
+
+		else:
+			return render_template('compare.html')
+
+
 	@app.route('/results')
 	def results():
-		# print(session)
+		return render_template(results.html)
+
+	# Present_id page (generates session id or code)
+	@app.route('/present_id')
+	def present_id():
+		person = {}
 		if 'stock_dict' in session:
 			person = algorithm.generate_profile(session['stock_dict'], finnhub_client)
 			print(person)
 		else:
 			return redirect(url_for('index'))
 		if 'code' in session:
-			return render_template('results.html', code=session['code'], warning=True)
+			return render_template('present_id.html', code=session['code'], warning=True)
```

Comment:
I realized that there's a small bug here (I think it's a bug at least?). If a session code has been generated and saved to the Flask session, any subsequent attempts to generate a profile will not be saved to the SQLite database due to this if condition, since the session code is currently never popped from the Flask session. This means that in order for a user to upload a different portfolio or update a previously uploaded portfolio, they would need to clear their cookies in order to remove the Flask session object (which is probably not what we want). I'll push a fix to this issue but if this is actually the behavior that we want then feel free to revert my changes.